psst-gui/src/delegate.rs

```diff
@@ -128,8 +128,12 @@ impl Delegate {
                     .unwrap();
             } else {
                 self.spawn(move || {
+                    let image_format = match &location[location.rfind(".").unwrap_or(0)..] {
```

Comment:
Maybe use endswith https://doc.rust-lang.org/std/string/struct.String.html#method.ends_with? Alternatively if its a filepath, extension https://doc.rust-lang.org/std/path/struct.Path.html#method.extension

Reply:
Also rather than using a match for a single case, you can use an `if let` expression https://doc.rust-lang.org/rust-by-example/flow_control/if_let.html